lib/ddtrace/contrib/delayed_job/patcher.rb

```diff
@@ -0,0 +1,50 @@
+module Datadog
+  module Contrib
+    module DelayedJob
+      # DelayedJob integration
+      module Patcher
+        include Base
+        register_as :delayed_job
+
+        option :service_name, default: 'delayed_job'.freeze
+        option :tracer, default: Datadog.tracer
+
+        @patched = false
+
+        class << self
+          def patch
+            return @patched if patched? || !defined?(::Delayed)
+
+            require 'ddtrace/ext/app_types'
+            require_relative 'plugin'
+
+            add_instrumentation(::Delayed::Worker)
+            add_pin(::Delayed::Worker)
+            @patched = true
+          rescue => e
+            Tracer.log.error("Unable to apply DelayedJob integration: #{e}")
+            @patched
+          end
+
+          def patched?
+            @patched
+          end
+
+          private
+
+          def unpatch
+            @patched = false
+          end
+
+          def add_instrumentation(klass)
+            klass.plugins << Plugin
+          end
+
+          def add_pin(klass)
```

Comment:
We're driving out configuration off the configuration API, which is good, but why are we adding a pin here?